bundles/org.openhab.binding.fsinternetradio/src/main/java/org/openhab/binding/fsinternetradio/internal/FSInternetRadioDiscoveryParticipant.java

```diff
@@ -41,11 +45,13 @@
  * @author Patrick Koenemann - Initial contribution
  * @author Mihaela Memova - removed the getLabel(RemoteDevice device) method due to its unreachable code lines
  */
+@NonNullByDefault
 @Component(immediate = true)
 public class FSInternetRadioDiscoveryParticipant implements UpnpDiscoveryParticipant {
+    private final Logger logger = LoggerFactory.getLogger(FSInternetRadioDiscoveryParticipant.class);
 
     /** Map from UPnP manufacturer to model number for supported radios; filled in static initializer below. */
-    private static final Map<String, Set<String>> SUPPORTED_RADIO_MODELS = new HashMap<String, Set<String>>();
+    private static final Map<@Nullable String, Set<@Nullable String>> SUPPORTED_RADIO_MODELS = new HashMap<@Nullable String, Set<@Nullable String>>();
```

Comment:
```suggestion
    private static final Map<String, Set<@Nullable String>> SUPPORTED_RADIO_MODELS = new HashMap<>();
```

keys are not allowed to be null. Why do you need the `@Nullable` in the Set? Is this to suppress the nullable warning on the get? 

Also you can use the diamond operator (this applies below, e.g. l. 63, 84, ...).

Reply:
I started to look to Optional, but not yet familiar with it. Could you provide 1-1 examples?

What is the diamond operator?

Reply:
I started to look to Optional, but not yet familiar with it. Could you provide 1-1 examples?

What is the diamond operator?

Reply:
```
private Map<String, WhateverType> myMap = new HashMap<String, WhateverType>();
private Map<String, WhateverType$ myMap = new HashMap<>();
```

First one is without, second one is with diamond operator (`<>`). You can omit the type for the initialization if you have declared it before.

Regarding the Optionals: I think this cannot be explained in some lines. I must admit that I still learn something new there. It‘s quite different in thinking and a lot of things cannot be changed 1:1y

Reply:
I fixed usage of the diamond operator
I removed @NonNullByDefault, this needs more time